### PR TITLE
kernel: remove legacy fields in _kernel

### PR DESCRIPTION
**Pull request by tyrantlizard77**
UP should just use _kernel.cpus[0].

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

### CODE REVIEW

**Review by carlescufi — COMMENTED** [review]
No objections, thanks.

**Review by carlescufi — APPROVED** [review]

**Review by wentongwu — APPROVED** [review]

File: arch/posix/core/swap.c [hunk 1]
```diff
@@ -22,16 +22,16 @@
 int arch_swap(unsigned int key)
 {
 	/*
-	 * struct k_thread * _kernel.current is the currently runnig thread
+	 * struct k_thread * _current is the currently runnig thread
```
Comment (ioannisg) [inline arch/posix/core/swap.c]: can you fix the typo in "runnig"?

**Review by aescolar — APPROVED** [review]
No complaints in posix arch/board related changes

**Comment by andrewboie** [comment]
@andyross ping

File: arch/posix/core/swap.c (hunk 1 above)
Comment (ioannisg) [reply arch/posix/core/swap.c:25]: @andrewboie ^^

**Review by ruuddw — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**